### PR TITLE
Feature/calendar

### PR DESCRIPTION
**Pull request by TawfiqKhan**
### What this PR does (required):
- Adds a Calendar component
- Event Cant be dragged
- event can be selected which will open up a modal with relevant information
- event title, the description can be edited
- placeholder button for image upload and delete

### Screenshots / Videos (required):
A5E8154A-C7A3-46E2-81D9-2EFF99F8341E_1_105_c
48271379-3280-45CF-9810-DC6E4579B424_1_105_c

### Any information needed to test this feature (required):
- install required packages

### Any issues with the current functionality (optional):
- add some Temporary code in Dashboard.tsx to switch views between board and Calendar by clicking the button on the nav We will probably add a more robust Tab feature in the future.

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -40,12 +41,13 @@ function App(): JSX.Element {
           <Switch>
             <Route exact path="/login" component={Login} />
             <Route exact path="/signup" component={Signup} />
+            {/* <Route exact path="/calendar" component={MyCalendar} /> */}
```
Comment (moffatethan) [inline client/src/App.tsx:44]: Why do we need to comment this out?

File: client/src/App.tsx [hunk 2]
```diff
@@ -40,12 +41,13 @@ function App(): JSX.Element {
...
             <Route exact path="/dashboard">
               <Dashboard />
             </Route>
-            <Route path="*">
+            {/* <Route path="*">
               <Redirect to="/dashboard" />
-            </Route>
+            </Route> */}
```
Comment (moffatethan) [inline client/src/App.tsx:50]: Same here, we should try making sure all the routes work with the feature so we aren't commenting important parts of the app out.

File: client/src/pages/Calendar/CardModal.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+          <DescriptionOutlined color="action" />
+          <Typography className={classes.header} variant="h5" component="h3">
+            Description
+          </Typography>
+        </Box>
+        <form className={classes.form} noValidate autoComplete="off">
+          <TextField
+            id="outlined-basic"
+            onFocus={handleFocus}
+            multiline
```
Comment (moffatethan) [inline client/src/pages/Calendar/CardModal.tsx:77]: Could we utilise Formik here?

File: client/src/pages/Calendar/CardModal.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                <ControlPoint color="action" />
+              </Button>
+            </label>
+          </Box>
+        </Box>
+
+        <Box className={classes.images}>
+          {content.images.map((image: string, index: number) => {
+            return (
+              <div className={classes.imageContainer} key={index}>
```
Comment (moffatethan) [inline client/src/pages/Calendar/CardModal.tsx:115]: It's actually an anti-pattern to do index as keys in React, I would utilise something like uuid or nanoid to follow best practice.

**Review by moffatethan — COMMENTED** [review]
Looks good, but left some comments for you to address

File: client/src/App.tsx (hunk 1 above)
Comment (TawfiqKhan) [reply client/src/App.tsx:44]: I have removed the the route, we are displaying the calendar as a tab.

File: client/src/App.tsx (hunk 2 above)
Comment (TawfiqKhan) [reply client/src/App.tsx:50]: fixed.

**Review by moffatethan — APPROVED** [review]
Good improvement! Thank you!

**Comment by moffatethan** [comment]
Please address merge conflicts before merging

**Pull request merged**